streams/src/main/java/org/apache/kafka/streams/kstream/internals/StreamJoinedInternal.java

```diff
@@ -18,16 +18,30 @@
 package org.apache.kafka.streams.kstream.internals;
 
 import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.TopologyConfig;
 import org.apache.kafka.streams.kstream.StreamJoined;
+import org.apache.kafka.streams.state.DslStoreSuppliers;
 import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
 
 import java.util.Map;
 
 public class StreamJoinedInternal<K, V1, V2> extends StreamJoined<K, V1, V2> {
 
+    private final InternalStreamsBuilder builder;
+
     //Needs to be public for testing
-    public StreamJoinedInternal(final StreamJoined<K, V1, V2> streamJoined) {
+    public StreamJoinedInternal(
+        final StreamJoined<K, V1, V2> streamJoined,
+        final InternalStreamsBuilder builder
+    ) {
         super(streamJoined);
+        this.builder = builder;
+        if (dslStoreSuppliers == null) {
+            final TopologyConfig topologyConfig = builder.internalTopologyBuilder.topologyConfigs();
+            if (topologyConfig != null) {
+                dslStoreSuppliers = topologyConfig.resolveDslStoreSuppliers().orElse(null);
+            }
```

Comment:
Hm...is this logic correct in the context of the initialization logic in [OuterStreamJoinedStoreFactory](https://github.com/apache/kafka/pull/14648/files#r1399812871)

My read of that code, specifically case 1, was that we interpret `streamJoined#dslStoreSuppliers` as what the user passed in to the StreamJoined, so it should be null (and stay null) if the user didn't explicitly configure a dslStoreSuppliers.

Then in case 3 (or the else branch of case 2), we fall back to the config value if neither the `dslStoreSuppliers` or the `thisStoreSupplier` of the StreamJoined was set/non-null

Reply:
nice catch! added a test and fixed it